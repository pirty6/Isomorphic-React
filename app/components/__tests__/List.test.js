import React from 'react';
import { shallow } from 'enzyme';

import List from '../List';

const mockPokemon = [
    {
        pokemon: {
            name: 'friendly'
        }
    }
];

const mockLocation = {
    match: {
        params: {
            ability: 'swagger'
        }
    }
}

describe('List', () => {
    it('should render correctly when given location', () => {
        const mockPokemon = [
            {
                pokemon: {
                    name: 'friendly'
                }
            }
        ];
        const mockLocation = {
            match: {
                params: {
                    ability: 'swagger'
                }
            }
        }
        const component = shallow(<List pokemon={mockPokemon} location={mockLocation} />);
        expect(component).toMatchSnapshot();
    })

    it('should render correctly when not given location', () => {
        const mockPokemon = [
            {
                pokemon: {
                    name: 'friendly'
                }
            }
        ];
        const mockLocation = {
            match: {
                params: {
                    ability: 'swagger'
                }
            }
        }
        const component = shallow(<List pokemon={mockPokemon} location={mockLocation} />);
        expect(component).toMatchSnapshot();
    })
})
