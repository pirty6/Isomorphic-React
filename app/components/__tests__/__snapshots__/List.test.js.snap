// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly when given location 1`] = `
<div>
  <h3>
    swagger
  </h3>
  <ul>
    <li>
      friendly
    </li>
  </ul>
</div>
`;

exports[`List should render correctly when not given location 1`] = `
<div>
  <h3>
    swagger
  </h3>
  <ul>
    <li>
      friendly
    </li>
  </ul>
</div>
`;
